cylc/flow/scheduler.py

```diff
@@ -902,11 +902,35 @@ def get_command_method(self, command_name: str) -> Callable:
         """Return a command processing method or raise AttributeError."""
         return getattr(self, f'command_{command_name}')
 
-    def queue_command(self, command: str, kwargs: dict) -> None:
-        self.command_queue.put((
-            command,
-            tuple(kwargs.values()), {}
-        ))
+    def queue_command(
+        self,
+        name: str,
+        args: list,
+        kwargs: dict
+    ) -> str:
+        """Queue a command for action by the scheduler.
+
+        Return a unique command ID for provenance tracking.
```

Comment:
:heart:

Reply:
(although we should really be setting this at the client rather than the server so we have one unique ID for the entire transaction)

Reply:
For future work we should set this at the client rather than the server so we have one unique ID with which to track the whole transaction. This ID could be used to prevent duplicate events being fired off by mistake and things like that.

Reply:
Agreed, I'll put up an issue for that.